pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java

```diff
@@ -630,8 +630,10 @@ private boolean copyInputStreamToDst(InputStream inputStream, URI dstUri, byte[]
     int bytesRead;
     long totalBytesRead = 0;
     byte[] buffer = new byte[BUFFER_SIZE];
+    // TODO: the newer client now has the API 'uploadFromFile' that directly takes the file as an input. We can replace
```

Comment:
Do we need to upgrade Azure dependency to fix this TODO?

Reply:
we don't need to bump up the dependency for this. This just needs a bit larger surgery of the adls gen 2 client (rewriting the copyFromLocal / copyToLocal). I will probably file another pr for doing that.